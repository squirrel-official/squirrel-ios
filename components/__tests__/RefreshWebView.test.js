/**
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 */

import { render } from '@testing-library/react-native';
import React from 'react';

import RefreshWebView from '../RefreshWebView';

// NOTE: This test just verifies the component renders, because the
// functionality would be very difficult to test properly
describe('RefreshWebView', () => {
	it('should render', () => {
		const { toJSON } = render(
			<RefreshWebView
				isRefreshing={false}
			/>
		);

		expect(toJSON()).toMatchSnapshot();
	});
});
