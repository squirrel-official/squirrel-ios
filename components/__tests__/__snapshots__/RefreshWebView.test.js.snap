// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RefreshWebView should render 1`] = `
<RCTScrollView
  collapsable={false}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onLayout={[Function]}
  refreshControl={
    <RefreshControlMock
      enabled={false}
      refreshing={false}
    />
  }
  showsHorizontalScrollIndicator={false}
  showsVerticalScrollIndicator={false}
  style={
    Object {
      "flex": 1,
      "height": "100%",
    }
  }
>
  <RCTRefreshControl />
  <View>
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "overflow": "hidden",
          },
          undefined,
        ]
      }
    >
      <RNCWebView
        cacheEnabled={true}
        injectedJavaScriptBeforeContentLoadedForMainFrameOnly={true}
        injectedJavaScriptForMainFrameOnly={true}
        javaScriptEnabled={true}
        messagingEnabled={false}
        onContentProcessDidTerminate={[Function]}
        onHttpError={[Function]}
        onLoadingError={[Function]}
        onLoadingFinish={[Function]}
        onLoadingProgress={[Function]}
        onLoadingStart={[Function]}
        onMessage={[Function]}
        onScroll={[Function]}
        onShouldStartLoadWithRequest={[Function]}
        source={null}
        style={
          Array [
            Object {
              "flex": 1,
              "overflow": "hidden",
            },
            Object {
              "backgroundColor": "#ffffff",
            },
            Object {
              "flex": 1,
              "height": 1334,
            },
          ]
        }
        textInteractionEnabled={true}
        useSharedProcessPool={true}
      />
    </View>
  </View>
</RCTScrollView>
`;
